qemu/softmmu/memory.c

```diff
@@ -79,6 +82,48 @@ MemoryRegion *memory_map_ptr(struct uc_struct *uc, hwaddr begin, size_t size, ui
     return ram;
 }
 
+static void make_contained(struct uc_struct *uc, MemoryRegion *current)
+{
+    hwaddr addr = current->addr;
+    MemoryRegion *container = g_new(MemoryRegion, 1);
+    memory_region_init(uc, container, int128_get64(current->size));
+    memory_region_del_subregion(uc->system_memory, current);
+    memory_region_add_subregion_overlap(container, 0, current, current->priority);
```

Comment:
Just a reminder, this could slow down normal read/write operations though should be not significant in your case (fuzzing).

Reply:
I have not micro-benchmarked it but it's amazing fast. Also keep in mind this is done only on CoW and only one level deep. The `FlatView` does the rest.